crates/threshold-signature-server/src/validator/api.rs

```diff
@@ -360,3 +359,22 @@ pub async fn prune_old_holders(
         validators_info.clone()
     })
 }
+
+/// Checks if TSS is a proper signer and if isn't deletes their parent key if they have one
+pub async fn is_proper_signer(
+    account_id: &AccountId32,
+    validators_info: Vec<ValidatorInfo>,
+    kv_manager: &KvManager,
+) -> Result<bool, ValidatorErr> {
+    let is_proper_signer =
+        &validators_info.iter().any(|validator_info| validator_info.tss_account == *account_id);
+    if *is_proper_signer {
+        Ok(true)
+    } else {
```

Comment:
```suggestion
    let is_proper_signer =
        validators_info.iter().any(|validator_info| validator_info.tss_account == *account_id);
    if is_proper_signer {
        Ok(true)
    } else {
```

Don't need the reference here